src/app/ui/components/Account/AccountDetails/BonusPlaces/BonusPlacesList.py

```
@@ -14,113 +14,188 @@ def __init__(self, parent, account_id, challenge_id):
+                    child_layout.deleteLater()
+
+    def load_bonus_places(self):
+        try:
+            completed_places = self.db_handler.get_all_completed_bonus_places_for_account(
+                self.account_id, self.challenge_id)
+            self.completed_ids = {place["bonus_place_id"] for place in completed_places}
+
+            self.all_bonus_places = self.db_handler.get_all_bonus_places(self.challenge_id)
+
+            self.process_timestamps()
+        except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)